internal/provider/sdkv2/tags_interceptor.go

```diff
@@ -297,3 +298,54 @@ func setTagsAll() customizeDiffInterceptor {
 		return nil
 	})
 }
+
+func validateRequiredTags() customizeDiffInterceptor {
+	return interceptorFunc1[*schema.ResourceDiff, error](func(ctx context.Context, opts customizeDiffInterceptorOptions) error {
+		c := opts.c
+
+		switch d, when, why := opts.d, opts.when, opts.why; when {
+		case Before:
+			switch why {
+			case CustomizeDiff:
+				isCreate := d.GetRawState().IsNull()
+				hasTagsChange := d.HasChange(names.AttrTags)
+
+				if isCreate || hasTagsChange {
```

Comment:
Similar to framework suggestion, I would flip this to return early pattern to make it easier to read, less indented.